File: PalindromeCheck.java
import java.util.Scanner;

public class PalindromeCheck {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String str = sc.nextLine();

        // Convert to lowercase to make it case-insensitive
        str = str.toLowerCase();

        // Reverse the string
        String reversed = new StringBuilder(str).reverse().toString();

        // Check if original equals reversed
        if (str.equals(reversed))
            System.out.println("It is a palindrome.");
        else
            System.out.println("It is not a palindrome.");
    }
}

File: PalindromeCheck.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main() {
    char str[100];
    printf("Enter a string: ");
    gets(str);

    // Convert to lowercase
    for (int i = 0; str[i]; i++) {
        str[i] = tolower(str[i]);
    }

    int len = strlen(str);
    int flag = 1;

    for (int i = 0; i < len / 2; i++) {
        if (str[i] != str[len - i - 1]) {
            flag = 0;
            break;
        }
    }

    if (flag)
        printf("It is a palindrome.\n");
    else
        printf("It is not a palindrome.\n");

    return 0;
}
